lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -74,21 +74,21 @@ module Deserialization
         #
         def parse!(document, options = {})
           parse(document, options) do |invalid_payload, reason|
-            fail InvalidDocument, "Invalid payload (#{reason}): #{invalid_payload}"
+            fail JSONAPI::Parser::InvalidDocument, "Invalid payload (#{reason}): #{invalid_payload}"
           end
         end
 
         # Same as parse!, but returns an empty hash instead of raising InvalidDocument
         # on invalid payloads.
         def parse(document, options = {})
-          document = document.dup.permit!.to_h if document.is_a?(ActionController::Parameters)
-
-          validate_payload(document) do |invalid_document, reason|
-            yield invalid_document, reason if block_given?
-            return {}
-          end
+          document = JSONAPI.parse(document, options)
+          document = document.to_hash
```

Comment:
The validator is as quick as it could possibly be (given it is written in ruby, but then again so is rails so I doubt it is a bottleneck).
Would you mind opening an issue on jsonapi-deserializable?